server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -1434,6 +1434,14 @@ public final boolean shouldProcessCheckpoint(ReplicationCheckpoint requestCheckp
             logger.trace(() -> new ParameterizedMessage("Ignoring new replication checkpoint - shard is not started {}", state()));
             return false;
         }
+        if (getReplicationTracker().isPrimaryMode()) {
+            logger.warn(
```

Comment:
Nitpick - You don't really need lazy logging here since:

1. `warn` is a pretty high log level and is unlikely to be suppressed
2. The actual log line is a simple string and doesn't involve any heavyweight object dumps

Reply:
Yes makes sense, I will remove lazy loading before merging in next commit.